hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/autocleanup/AutoActionCleanupTest.java

```diff
@@ -0,0 +1,227 @@
+
+        autoActionCleanup.run();
+
+        assertThat(actionRepository.count()).isEqualTo(2);
+
+    }
+
+    @Test
+    @Description("Verifies that nothing is cleaned up if the cleanup is disabled.")
+    public void cleanupDisabled() {
+
+        // cleanup config for this test case
+        setupCleanupConfiguration(false, 0, Action.Status.CANCELED);
+
+        final Target trg1 = testdataFactory.createTarget("trg1");
+        final Target trg2 = testdataFactory.createTarget("trg2");
+
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("ds1");
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("ds2");
+
+        final Long action1 = deploymentManagement.assignDistributionSet(ds1.getId(), ActionType.FORCED, 0,
+                Collections.singletonList(trg1.getControllerId())).getActions().get(0);
+        deploymentManagement.assignDistributionSet(ds2.getId(), ActionType.FORCED, 0,
+                Collections.singletonList(trg2.getControllerId()));
+
+        setActionToCanceled(action1);
+
+        assertThat(actionRepository.count()).isEqualTo(2);
+
+        autoActionCleanup.run();
+
+        assertThat(actionRepository.count()).isEqualTo(2);
+
+    }
+
+    @Test
+    @Description("Verifies that canceled and failed actions are cleaned up.")
+    public void canceledAndFailedActionsAreCleanedUp() {
+
+        // cleanup config for this test case
+        setupCleanupConfiguration(true, 0, Action.Status.CANCELED, Action.Status.ERROR);
+
+        final Target trg1 = testdataFactory.createTarget("trg1");
+        final Target trg2 = testdataFactory.createTarget("trg2");
+        final Target trg3 = testdataFactory.createTarget("trg3");
+
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("ds1");
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("ds2");
+
+        final Long action1 = deploymentManagement.assignDistributionSet(ds1.getId(), ActionType.FORCED, 0,
+                Collections.singletonList(trg1.getControllerId())).getActions().get(0);
+        final Long action2 = deploymentManagement.assignDistributionSet(ds2.getId(), ActionType.FORCED, 0,
+                Collections.singletonList(trg2.getControllerId())).getActions().get(0);
+        final Long action3 = deploymentManagement.assignDistributionSet(ds2.getId(), ActionType.FORCED, 0,
+                Collections.singletonList(trg3.getControllerId())).getActions().get(0);
+
+        assertThat(actionRepository.count()).isEqualTo(3);
+
+        setActionToCanceled(action1);
+        setActionToFailed(action2);
+
+        assertThat(actionRepository.count()).isEqualTo(3);
+
+        autoActionCleanup.run();
+
+        assertThat(actionRepository.count()).isEqualTo(1);
+        assertThat(actionRepository.getById(action3)).isPresent();
+
+    }
+
+    @Test
+    @Description("Verifies that canceled and failed actions are cleaned up.")
```

Comment:
Please adapt description. This is `Canceled`only.  ... would it also make sense, to provide a test for `Failed`only?

Reply:
Done